Transfer/Sources/TransferAmountCell.swift

```diff
@@ -0,0 +1,159 @@
+
+    var enteredAmountHandler: EnteredAmountHandler?
+
+    lazy var amountTextField: PasteOnlyTextField = {
+        let textField = PasteOnlyTextField(frame: .zero)
+        textField.textAlignment = .right
+        textField.keyboardType = UIKeyboardType.numberPad
+        textField.delegate = self
+        textField.translatesAutoresizingMaskIntoConstraints = false
+        textField.accessibilityIdentifier = "transferAmountTextField"
+        return textField
+    }()
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: .value1, reuseIdentifier: reuseIdentifier)
+        self.selectionStyle = .none
+        let tap = UITapGestureRecognizer(target: self, action: #selector(didTapCell))
+        self.addGestureRecognizer(tap)
+
+        setupAmountTextField()
+    }
+
+    @objc
+    private func didTapCell() {
+        amountTextField.becomeFirstResponder()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+
+    private func setupAmountTextField() {
+        textLabel?.accessibilityIdentifier = "transferAmountTitleLabel"
+        detailTextLabel?.accessibilityIdentifier = "transferAmountCurrencyLabel"
+        contentView.addSubview(amountTextField)
+        amountTextField.safeAreaCenterYAnchor
+            .constraint(equalTo: contentView.layoutMarginsGuide.centerYAnchor).isActive = true
+        amountTextField.safeAreaTrailingAnchor
+            .constraint(equalTo: detailTextLabel!.layoutMarginsGuide.leadingAnchor, constant: -15).isActive = true
+        let amountTextFieldLeadingConstraint = amountTextField.safeAreaLeadingAnchor
+            .constraint(equalTo: textLabel!.layoutMarginsGuide.trailingAnchor, constant: 15)
+        amountTextFieldLeadingConstraint.priority = UILayoutPriority(999)
+        amountTextFieldLeadingConstraint.isActive = true
+    }
+
+    func configure(amount: String?, currency: String?, isEnabled: Bool, _ handler: @escaping EnteredAmountHandler) {
+        textLabel?.text = "transfer_amount".localized()
+        amountTextField.text = amount
+        amountTextField.isEnabled = isEnabled
+        detailTextLabel?.text = currency
+        enteredAmountHandler = handler
+    }
+}
+
+extension TransferAmountCell: UITextFieldDelegate {
+    private static var amountFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.maximumFractionDigits = 2
+        formatter.minimumFractionDigits = 2
+        formatter.minimumIntegerDigits = 1
+        formatter.numberStyle = .currency
+        formatter.currencyCode = ""
+        formatter.currencySymbol = ""
+        formatter.locale = Locale(identifier: Locale.preferredLanguages[0])
+        return formatter
+    }()
+
+    func textFieldDidEndEditing(_ textField: UITextField) {
+        enteredAmountHandler?(amountTextField.text ?? "")
+    }
+
+    func textField(_ textField: UITextField,
```

Comment:
what is this function for?

Reply:
for this:
    entering a 9, results in 0.09
    entering a 4 after the 9 results in 0.94
    entering a 2 after the 4 results in 9.42
    entering a 3 after the 4 results in 94.23
    etc.